frontend/src/components/common/PrintCSVButton.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useState } from "react";
+import {
+  Icon,
+  IconButton,
+  Box,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  FormControl,
+  Button,
+  FormHelperText,
+  Flex,
+  ScaleFade,
+  Alert,
+  AlertDescription,
+  AlertIcon,
+} from "@chakra-ui/react";
+import { AiFillPrinter } from "react-icons/ai";
+import { RangeDatepicker } from "chakra-dayzed-datepicker";
+import commonAPIClient from "../../APIClients/CommonAPIClient";
+import CSVConverter from "../../helper/CSVConverter";
+
+const PrintCSVButton = (): React.ReactElement => {
+  const [selectedDates, setSelectedDates] = useState<Date[]>([]);
+  const [showAlert, setShowAlert] = useState(false);
+
+  const handleClear = () => {
+    setSelectedDates([]);
+  };
+
+  const [isOpen, setOpen] = useState(false);
+
+  const handleOpen = () => {
+    setOpen(true);
+    setSelectedDates([]);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handleSubmit = async () => {
+    // Use YYYY-MM-DD format
+    const dateRange = selectedDates.map((date) =>
+      date.toISOString().substring(0, 10),
+    );
+
+    const data = await commonAPIClient.filterLogRecords(
+      undefined,
+      undefined,
+      undefined,
+      dateRange,
+      undefined,
+      undefined,
+      true, // return all data
+    );
+
+    setShowAlert(CSVConverter(data.logRecords) === "error");
```

Comment:
As an added safety check, just so we don't get any unwanted `Cannot read properties of null` errors

```suggestion
    setShowAlert(!data || CSVConverter(data.logRecords) === "error");
```

Reply:
This was a good catch, thanks!